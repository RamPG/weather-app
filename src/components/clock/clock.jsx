import React, { useState, useEffect } from 'react';

import { getHoursFormat, getMinutesFormat, getSecondsFormat } from '../../services/time-library';

const useClock = () => {
  const [hours, setHours] = useState(getHoursFormat());
  const [minutes, setMinutes] = useState(getMinutesFormat());
  const [seconds, setSeconds] = useState(getSecondsFormat());
  useEffect(() => {
    setInterval(() => {
      setHours(getHoursFormat());
      setMinutes(getMinutesFormat());
      setSeconds(getSecondsFormat());
    }, 1000);
  }, [hours, minutes, seconds]);
  return { hours, minutes, seconds };
};
const Clock = () => {
  const time = useClock();
  const { hours, minutes, seconds } = time;
  return (
    <h1>
      {hours}
      :
      {minutes}
      :
      {seconds}
    </h1>
  );
};

export default Clock;
