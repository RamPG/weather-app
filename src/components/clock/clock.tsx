import React, { FunctionComponent } from 'react';

import './clock.scss';

import {
  getHours, getMinutes, getSeconds,
} from '../../services/time-library';
import { useClock } from '../../user-hooks/use-clock';

type ClockPropsType = {
    hoursStart: number,
    minutesStart: number,
    secondsStart: number,
}

export const Clock: FunctionComponent<ClockPropsType> = (
  { hoursStart, minutesStart, secondsStart },
) => {
  const { hours, minutes } = useClock(hoursStart, minutesStart, secondsStart, 60000);
  return (
    <p>
      Time:
      {' '}
      {hours}
      :
      {minutes}
    </p>
  );
};
