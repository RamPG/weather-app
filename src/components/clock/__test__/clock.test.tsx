import React from 'react';
import { render, cleanup, queries } from '@testing-library/react';
import { Clock, ClockPropsType } from '../clock';

describe('Clock component', () => {
  let props: ClockPropsType;
  afterEach(cleanup);
  beforeAll(() => {
    props = {
      hoursStart: 12,
      minutesStart: 0,
      secondsStart: 0,
    };
  });
  it('Snapshot', () => {
    const { asFragment } = render(<Clock {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
