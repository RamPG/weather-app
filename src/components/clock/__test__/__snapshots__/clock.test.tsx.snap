// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock component test Clock component snapshot 1`] = `
<DocumentFragment>
  <p>
    Time: 12:00
  </p>
</DocumentFragment>
`;
